tests/test_reportlog.py

```diff
@@ -45,6 +51,22 @@ def test_fail():
         "$report_type": "SessionFinish",
     }
 
+    split = defaultdict(list)
+    for obj in json_objs:
+        split[obj["$report_type"] == "warning-recorded"].append(obj)
```

Comment:
Neat. 😁

Reply:
I would usually use `itertools.groupby` (or, even better, `more_itertools.bucket` or `toolz.itertoolz.groupby`) for this, but the bare `itertools.groupby` needs sorted sequences to work properly, making the code a bit too complicated. The `toolz` and `more_itertools` versions don't have that issue, but I'd be introducing a new test dependency for those.